sentry/src/main/java/io/sentry/clientreport/ClientReport.java

```diff
@@ -0,0 +1,132 @@
+package io.sentry.clientreport;
+
+import io.sentry.DateUtils;
+import io.sentry.ILogger;
+import io.sentry.JsonDeserializer;
+import io.sentry.JsonObjectReader;
+import io.sentry.JsonObjectWriter;
+import io.sentry.JsonSerializable;
+import io.sentry.JsonUnknown;
+import io.sentry.SentryLevel;
+import io.sentry.vendor.gson.stream.JsonToken;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class ClientReport implements JsonUnknown, JsonSerializable {
+
+  private final @NotNull Date timestamp;
+  private final @NotNull List<DiscardedEvent> discardedEvents;
+  private @Nullable Map<String, Object> unknown;
+
+  public ClientReport(@NotNull Date timestamp, @NotNull List<DiscardedEvent> discardedEvents) {
+    this.timestamp = timestamp;
+    this.discardedEvents = discardedEvents;
+  }
+
+  public @NotNull Date getTimestamp() {
+    return timestamp;
+  }
+
+  public @NotNull List<DiscardedEvent> getDiscardedEvents() {
+    return discardedEvents;
+  }
+
+  @Override
+  public String toString() {
+    return "ClientReport{"
+        + "timestamp="
+        + timestamp
+        + ", discardedEvents="
+        + discardedEvents
+        + '}';
+  }
```

Comment:
Usually, we don't override such methods because they just increase line numbers and final jar size.
Calling the logger with such values would do.

Reply:
Was used for better output while testing

Reply:
I'd remove it then and do it during testing only, can be an extension method on Kotlin.